Jenkinsfile

```diff
@@ -44,129 +23,130 @@ node('windows') {
+			}
 		}
 	}
+)
 
-	dir(workdir) {
-		stage "Checkout ${repo}"
-		git url: "https://github.com/deis/${repo}.git", branch: "master"
+stage 'Build and Upload CLI built with SDK'
+
+def gcs_bucket = "gs://workflow-cli"
+def wcli_image = 'quay.io/deisci/workflow-cli-dev:latest'
 
-		stage "Build ${repo}"
-		if (git_branch != "remotes/origin/master") {
-			echo "Skipping build of 386 binaries to shorten CI for Pull Requests"
-			env.BUILD_ARCH = "amd64"
-		}
-		make 'bootstrap'
 
-		stage "Update local glide.yaml with controller-sdk-go repo '${go_repo}' and version '${git_commit}'"
+def upload_artifacts = { String dist_dir ->
+	headers  = "-h 'x-goog-meta-git-branch:${git_branch}' "
+	headers += "-h 'x-goog-meta-git-sha:${git_commit}' "
+	headers += "-h 'x-goog-meta-ci-job:${env.JOB_NAME}' "
+	headers += "-h 'x-goog-meta-ci-number:${env.BUILD_NUMBER}' "
+	headers += "-h 'x-goog-meta-ci-url:${env.BUILD_URL}'"
 
-		def pattern = "github\\.com\\/deis\\/controller-sdk-go\\n\\s+version:\\s+[a-f0-9]+"
-		def replacement = "${go_repo.replace("/", "\\/")}\\n  version: ${git_commit}"
-		sh "perl -i -0pe 's/${pattern}/${replacement}/' glide.yaml"
+	script = "sh -c 'echo \${GCS_KEY_JSON} | base64 -d - > /tmp/key.json "
+	script += "&& gcloud auth activate-service-account -q --key-file /tmp/key.json "
+	script += "&& gsutil -mq ${headers} cp -a public-read -r /upload/* ${gcs_bucket} "
+	script += "&& rm -rf /upload/*'"
 
-		def glideYaml = readFile('glide.yaml')
-		echo "Updated glide.yaml:\n${glideYaml}"
+	withCredentials([[$class: 'StringBinding',
+										credentialsId: '6561701c-b7b4-4796-83c4-9d87946799e4',
+										variable: 'GCSKEY']]) {
+		sh "docker run ${dist_dir} -e GCS_KEY_JSON=\"\${GCSKEY}\" --rm ${wcli_image} ${script}"
+	}
+}
 
-		make 'glideup'
-		env.REVISION = git_commit.take(7)
-		env.VERSION = "csdk-ci-" + git_commit.take(7)
-		make 'build-revision'
+def mktmp = {
+	// Create tmp directory to store files
+	sh 'mktemp -d > tmp_dir'
+	tmp = readFile('tmp_dir').trim()
+	echo "Storing binaries in ${tmp}"
+	sh 'rm tmp_dir'
+	return tmp
+}
 
-		stage "Deploy ${repo}"
-		upload_artifacts()
+node('linux') {
+	flags = ""
+	if (git_branch != "remotes/origin/master") {
+		echo "Skipping build of 386 binaries to shorten CI for Pull Requests"
+		flags += "-e BUILD_ARCH=amd64"
 	}
+
+	tmp_dir = mktmp()
+	dist_dir = "-e DIST_DIR=/upload -v ${tmp_dir}:/upload"
+
+	def pattern = "github\\.com\\/deis\\/controller-sdk-go\\n\\s+version:\\s+[a-f0-9]+"
+	replacement = "github\\.com\\/deis\\/controller-sdk-go\\n"
+	replacement += "  repo: https:\\/\\/github\\.com\\/${env.CHANGE_AUTHOR}\\/controller-sdk-go\\.git\\n"
+	replacement += "  vcs: git\\n"
+	replacement += "  version: ${git_commit}"
+
+	build_script = "sh -c 'perl -i -0pe \"s/${pattern}/${replacement}/\" glide.yaml "
+	build_script += "&& rm -rf glide.lock vendor/github.com/deis/controller-sdk-go "
+	build_script += "&& glide install "
```

Comment:
Not really. `glide install` without a lock file is the same as `glide up`. I guess I could not delete the lock file and run `glide up`, but I wanted to make sure no references to `github.com/deis/contorller-sdk-go` were left over in it.